opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/AbstractForeignSourceRepository.java

```diff
@@ -63,13 +63,22 @@ public AbstractForeignSourceRepository() {
     public Requisition importResourceRequisition(final Resource resource) throws ForeignSourceRepositoryException {
         Assert.notNull(resource);
 
-        LOG.debug("importing requisition from {}", resource);
+        LOG.debug("importing requisition from {}", stripCredentials(resource));
         final Requisition requisition = JaxbUtils.unmarshal(Requisition.class, resource);
         requisition.setResource(resource);
         save(requisition);
         return requisition;
     }
 
+    static String stripCredentials(final Object object) {
+        if (object == null) {
+            return null;
+        } else {
+            return String.valueOf(object).replaceAll("(username=)[^;&]*(;&)?", "$1***$2")
```

Comment:
Does this regex catch if the password happens be the last query parameter of the URL?

Reply:
the `?` at the end makes it optional, I think, if I'm reading it right...

Reply:
Added test.

Reply:
Ok?